chains/heads/tracker.go

```diff
@@ -217,6 +259,15 @@ func (t *tracker[HTH, S, ID, BLOCK_HASH]) Backfill(ctx context.Context, headWith
 			latestFinalized.BlockNumber(), headWithChain.BlockNumber(), t.htConfig.MaxAllowedFinalityDepth())
 	}
 
+	if !t.instantFinality() {
```

Comment:
nit: while you are here and if the change does not blows up the scope. Could you replace 
```
t.log.With("head_block_num", headWithChain.BlockNumber(),
			"latest_finalized_block_number", latestFinalized.BlockNumber()).
			Criticalf(errMsg)
```

with a less strict warning, as this is a normal case on chains with fast finality - we are fetching latest finalized after we've fetched latest head, so`headWithChain.BlockNumber() < latestFinalized.BlockNumber()`